### PR TITLE
feat(hydroflow_plus): use trybuild to compile subgraph binaries

### LINKED ISSUE #1307: Use trybuild-esque build process for Hydroflow+
To simplify the build process

### PR DESCRIPTION
**Pull request by shadaj**
Fixes #1307

### CODE REVIEW

**Comment by shadaj** [comment]
For now, to compile in release mode you must manually provide the appropriate rustc flags. I'm going to leave supporting Cargo profiles to a later PR, that requires much more work.

**Comment by MingweiSamuel** [comment]
> For now, to compile in release mode you must manually provide the appropriate rustc flags. I'm going to leave supporting Cargo profiles to a later PR, that requires much more work.

#1400

File: hydroflow_plus_test/examples/perf_compute_pi.rs [hunk 1]
```diff
@@ -2,20 +2,21 @@ use std::sync::Arc;
...
-use hydro_deploy::{Deployment, Host, HydroflowCrate};
-use hydroflow_plus_cli_integration::{DeployClusterSpec, DeployProcessSpec};
-use stageleft::RuntimeData;
+use hydro_deploy::{Deployment, Host};
+use hydroflow_plus_cli_integration::TrybuildHost;
 use tokio::sync::RwLock;
 
 type HostCreator = Box<dyn Fn(&mut Deployment) -> Arc<dyn Host>>;
 
+// TODO(shadaj): currently broken due to `profile` profile not being found
```
Comment (MingweiSamuel) [inline hydroflow_plus_test/examples/perf_compute_pi.rs:11]: !

**Review by MingweiSamuel — COMMENTED** [review]
exciting!

File: hydro_deploy/hydro_cli/src/lib.rs [hunk 2]
```diff
@@ -279,7 +279,10 @@ impl Deployment {
                 bin,
                 example,
                 profile,
-                None, // Python API doesn't support perf
+                None,  // Python API doesn't support rustflags
+                None,  // Python API doesn't support target_dir
+                false, // Python API doesn't support no_default_features
+                None,  // Python API doesn't support perf
```
[the remark this answers is not in the document]
Comment (shadaj) [reply hydro_deploy/hydro_cli/src/lib.rs:285]: I guess for now... not super sure if we care to maintain it going forward though. Probably a separate discussion to have.

**Review by MingweiSamuel — APPROVED** [review]

**Pull request merged**